### PR TITLE
[2166] Add support for attachment type configuration

### PR DESCRIPTION
**Pull request by bychkovdmitry**
### 🎯 Goal

Hide files from the attachment picker that are not valid according to the upload config.

### 🧪 Testing

1. Navigate to the "Upload Configuration" section in the dashboard
2. Experiment with allowed / blocked file extensions and MIME types
3. Open our attachment picker in the app and ensure that only files matching the filters in the dashboard are displayed

**Please, reset the upload configuration to the defaults after testing**

#### General
- [x] I have signed the [Stream CLA](https://docs.google.com/forms/d/e/1FAIpQLScFKsKkAJI7mhCr7K9rEIOpqIDThrWxuvxnwUq2XkHyG154vQ/viewform) (required)
- [x] Assigned a person / code owner group (required)
- [ ] Thread with the PR link started in a respective Slack channel (#android-chat-core or #android-chat-ui) (required)
- [ ] PR targets the `develop` branch
- [x] PR is linked to the GitHub issue it resolves

#### Code & documentation
- [x] Changelog is updated with client-facing changes
- [ ] New code is covered by unit tests
- [ ] Comparison screenshots added for visual changes
- [x] Affected documentation updated (KDocs, docusaurus, tutorial)

### ☑️Reviewer Checklist
- [ ] UI Components sample runs & works
- [ ] Compose sample runs & works
- [ ] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [ ] New feature tested and works
- [ ] Release notes and docs clearly describe changes
- [ ] All code we touched has new or updated KDocs

### CODE REVIEW

**Review by skydoves — APPROVED** [review]
Looks good to me!

### ☑️Reviewer Checklist
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [x] UI Changes correct (before & after images)
- [ ] Bugs validated (bugfixes)
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

File: CHANGELOG.md [hunk 1]
```diff
@@ -64,6 +64,7 @@ Added support for file upload configuration that lets you specify what types of
 ### ⬆️ Improved
 
 ### ✅ Added
+Added support for file upload configuration that let's you specify what types of files and images you want to allow or block from being uploaded. [3288](https://github.com/GetStream/stream-chat-android/pull/3288)
```
Comment (MarinTolic) [inline CHANGELOG.md:67]:
```suggestion
Added support for file upload configuration that lets you specify what types of files and images you want to allow or block from being uploaded. [3288](https://github.com/GetStream/stream-chat-android/pull/3288)
```

File: stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/util/StorageHelperWrapper.kt [hunk 2]
```diff
@@ -29,25 +30,32 @@ import io.getstream.chat.android.client.models.Attachment
...
+ * by the backend.
  */
 public class StorageHelperWrapper(
     private val context: Context,
     private val storageHelper: StorageHelper,
+    private val attachmentFilter: AttachmentFilter,
 ) {
 
     /**
      * Loads a list of file metadata from the system.
```
Comment (MarinTolic) [inline stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/util/StorageHelperWrapper.kt:43]: Maybe we just need to note in the kdocs for these methods that we filter the list against accepted mime types, but I'd leave this one up to your judgement.

Maybe something like:

```suggestion
     * Loads a list of file metadata from the system and filters it against file types accepted by the backend.
```

**Review by MarinTolic — APPROVED** [review]
- [x] UI Components sample runs & works
- [x] Compose sample runs & works
- [ ] ~~UI Changes correct (before & after images)~~ (no UI changes)
- [ ] ~~Bugs validated (bugfixes)~~
- [x] New feature tested and works
- [x] Release notes and docs clearly describe changes
- [x] All code we touched has new or updated KDocs

**Pull request merged**